third_party/amd/lib/TritonAMDGPUToLLVM/Utility.cpp

```diff
@@ -2,17 +2,42 @@
 #include "PatternTritonGPUOpToLLVM.h"
 #include "mlir/Dialect/LLVMIR/NVVMDialect.h"
 #include "mlir/Dialect/LLVMIR/ROCDLDialect.h"
+#include "triton/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVMBase.h"
 #include "triton/Conversion/TritonGPUToLLVM/TypeConverter.h"
 #include "triton/Dialect/NVGPU/IR/Dialect.h"
 
+using mlir::triton::gpu::appendOrGetExternFuncOp;
+using mlir::triton::gpu::getFunctionType;
+
 namespace {
 enum class ShflKind : uint32_t {
   bfly = 0,
   up = 1,
   down = 2,
   idx = 3,
 };
+
+std::string getTypeString(Type ty) {
```

Comment:
Is there a better way to translate a Type into a string ? This is hacky, but works :)